core/server/src/main/java/alluxio/worker/DefaultAlluxioWorker.java

```diff
@@ -96,13 +98,20 @@
   private long mStartTimeMs;
 
   /**
+   * The worker ID for this worker. This is set when the block worker is initialized and may be
+   * updated by the block sync thread if the master requests re-registration.
+   */
+  private AtomicReference<Long> mWorkerId;
+
+  /**
    * Constructs a {@link DefaultAlluxioWorker}.
    */
   public DefaultAlluxioWorker() {
+    mWorkerId = new AtomicReference<>();
```

Comment:
I found it strange as well, but that's how we've always done it and we would need a significant change to organize things more sensibly. The fundamental problem is that we use the same worker id for both the file system and block workers, but they are supposed to be completely independent. Not sure whether we want to use a different id for each, or have a meta-master/meta-worker which handle worker ids.